scripts/verify-release.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+RELEASE_TAG="$1"
+
+GITHUB_TOKEN="${GITHUB_TOKEN:-}"
+OWNER="${OWNER:-openbao}"
+REPO="${REPO:-openbao}"
+TMPDIR="${TMPDIR:-/tmp}"
+GPG_KEY_NAME="${GPG_KEY_NAME:-openbao-gpg-pub-20240618.asc}"
+
+DOWNLOAD_FOLDER="${TMPDIR}/${OWNER}-${REPO}-${RELEASE_TAG}"
+
+if [ ! -e "$DOWNLOAD_FOLDER" ]; then
+  mkdir -p "$DOWNLOAD_FOLDER"
+fi
+
+release_cache="${TMPDIR}/${OWNER}-${REPO}-release-${RELEASE_TAG}.json"
+release_api="https://api.github.com/repos/${OWNER}/${REPO}/releases/tags/${RELEASE_TAG}"
+bearer_header=""
+if [ -n "$GITHUB_TOKEN" ]; then
+  bearer_header="Authorization: Bearer $GITHUB_TOKEN"
+fi
+
+release_info=""
+if [ -e "$release_cache" ]; then
+  release_info="$(cat "$release_cache")"
+else
+  release_info="$(curl -sSL \
+    -H "Accept: application/vnd.github+json" \
+    -H "$bearer_header" \
+    -H "X-GitHub-Api-Version: 2022-11-28" \
+    "$release_api")"
+  echo "$release_info" > "$release_cache"
+fi
+
+assets_count="$(jq '.assets | length' <<< "$release_info")"
+
+for index in $(seq 0 $(( assets_count - 1 ))); do
+  download_url="$(jq -r ".assets[$index].browser_download_url" <<< "$release_info")"
+  download_file="$(basename "$download_url")"
+  download_path="${DOWNLOAD_FOLDER}/${download_file}"
+
+  if [ ! -e "$download_path" ]; then
+    echo "Downloading $download_file..."
+    curl -sSL "$download_url" --output "$download_path"
+    sleep 2
+  fi
+done
```

Comment:
Instead of looping over each individual asset here, it would be much quicker to make use of the `gh` CLI to download all assets in parallel.  For example:

```
gh release download v2.2.1 -D /tmp/openbao-openbao-v2.2.1/ --skip-existing
```

In general, making use of the GitHub CLI would also simplify other parts of the script, because we don't need to manually fetch all release assets first.

Reply:
Ah, I hadn't realized it has a downloader. I don't have it locally though... It looks like they don't maintain a container image either, so we'd have to build a custom Dockerfile for this. :laughing: 

https://github.com/cli/cli/issues/2027#issuecomment-977241407

Reply:
It is preinstalled on all GitHub Actions runners. I guess we're planning to run this script in CI anyways, so the missing image will not be an issue